File: design_patterns/java/src/structural/dapter/TextView.java
package structural.dapter;

/**
 * Adaptee， 已有的组件，可能是三方的库。能够显示想要的 Text 但是接口与现有的组件不兼容。
 */
public class TextView {
    private String content;

    public void setOrigin(float x, float y) {
        // ...
    }


    public void setExtent(float width, float height) {

    }


    boolean isEmpty() {
        return content == null || content.length() == 0;
    }
}

File: design_patterns/c++/src/structural/adapter/TextView.cpp
//
// Created by Albert on 1/3/21.
//

#include <clocale>
#include "../../../headers/structural/adapter/TextView.h"

TextView::TextView() = default;

// 接口不匹配体现在通过原点和宽，高来定会位置的边界。
void TextView::setOrigin(float x, float y) const {
  // do set
  // ...
}

// 宽， 高
void TextView::setExtent(float width, float height) const {
  // record the width and height.
}

bool TextView::isEmpty() const {
  return _content==NULL || _content.length==0;

}